### PR TITLE
Iosxe/show ap led brightness level summary

### PR DESCRIPTION
**Pull request by itdependsnetworks**
Would not merge until https://github.com/CiscoTestAutomation/genieparser/pull/252

### CODE REVIEW

File: src/genie/libs/parser/iosxe/show_ap.py [hunk 1]
```diff
@@ -295,3 +295,680 @@ def cli(self, output=None):
...
+
+
+# ===============================
+# Schema for:
+#  * 'show ap dot11 5ghz summary'
+# ===============================
+class ShowApDot115GhzSummarySchema(MetaParser):
+    """Schema for show ap dot11 5ghz summary."""
+
+    schema = {
```
Comment (xiaoxinz-cisco) [inline src/genie/libs/parser/iosxe/show_ap.py:680]: @tahigash please check the schema

File: src/genie/libs/parser/iosxe/show_ap.py [hunk 2]
```diff
@@ -295,3 +295,680 @@ def cli(self, output=None):
...
+
+        # AP Name                           Mac Address     Slot    Admin State    Oper State    Width  Txpwr           Channel
+        ap_header_capture = re.compile(
+            r"^AP\s+Name\s+Mac\s+Address\s+Slot\s+Admin\s+State\s+Oper\s+State\s+Width\s+Txpwr\s+Channel$")
+        # ---------------------------------------------------------------------------------------------------------------------------------
+        delimiter_capture = re.compile(
+            r"^---------------------------------------------------------------------------------------------------------------------------------$")
+        # ab22-cap10                    5c50.1504.8be0  1       Enabled        Up            20     *6/8 (9 dBm)    (132)*
+        ap_info_capture = re.compile(
+            r"^(?P<ap_name>\S+)\s+(?P<mac_address>\S+)\s+(?P<slot>\d+)\s+(?P<admin_state>(Enabled|Disabled))\s+(?P<oper_state>\S+)\s+(?P<width>\d+)\s+(?P<tx_pwr>\*.*dBm\))\s+(?P<channel>\S+)$")
```
Comment (xiaoxinz-cisco) [inline src/genie/libs/parser/iosxe/show_ap.py:745]: can you split this line 

https://www.python.org/dev/peps/pep-0008/#:~:text=Limit%20all%20lines%20to%20a,be%20limited%20to%2072%20characters.

File: src/genie/libs/parser/iosxe/show_ap.py [hunk 3]
```diff
@@ -295,3 +295,680 @@ def cli(self, output=None):
...
+
+
+# ===============================
+# Schema for:
+#  * 'show ap rf-profile summary'
+# ===============================
+class ShowApRfProfileSummarySchema(MetaParser):
+    """Schema for show ap rf-profile summary."""
+
+    schema = {
```
Comment (xiaoxinz-cisco) [inline src/genie/libs/parser/iosxe/show_ap.py:789]: @tahigash please check the schema

File: src/genie/libs/parser/iosxe/show_ap.py [hunk 4]
```diff
@@ -295,3 +295,680 @@ def cli(self, output=None):
...
+
+
+# =========================================
+# Schema for:
+#  * 'show ap led-brightness-level summary'
+# =========================================
+class ShowApLedBrightnessLevelSummarySchema(MetaParser):
+    """Schema for show ap led-brightness-level summary."""
+
+    schema = {
```
Comment (xiaoxinz-cisco) [inline src/genie/libs/parser/iosxe/show_ap.py:895]: @tahigash please check the schema

**Review by xiaoxinz-cisco — CHANGES_REQUESTED** [review]

File: src/genie/libs/parser/iosxe/tests/ShowApDot115GhzChannel/cli/equal/golden_output_expected.py [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+        "channel_interference": "Enable",
+        "channel_load": "Disable",
+        "device_aware": "Disable",
+        "clean_air": "Disabled",
+        "wlc_leader_name": "sj-00a-ewlc1",
+        "wlc_leader_ip": "10.7.5.133",
+        "last_run_seconds": 15995,
+        "dca_level": "MEDIUM",
+        "dca_db": 15,
+        "chan_width": "80 MHz",
```
Comment (tahigash) [inline src/genie/libs/parser/iosxe/tests/ShowApDot115GhzChannel/cli/equal/golden_output_expected.py:16]: let's have only number as same as line 17.
```
"chan_width_mhz": 80,

**Review by tahigash — CHANGES_REQUESTED** [review]
reviewed schema and left comments

File: src/genie/libs/parser/iosxe/tests/ShowApDot115GhzSummary/cli/equal/golden_output_expected.py [hunk 6]
```diff
@@ -0,0 +1,184 @@
+expected_output = {
```
Comment (tahigash) [inline src/genie/libs/parser/iosxe/tests/ShowApDot115GhzSummary/cli/equal/golden_output_expected.py:1]: schema looks good

File: src/genie/libs/parser/iosxe/tests/ShowApLedBrightnessLevelSummary/cli/equal/golden_output_output.txt [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+    b802-cap4                     8
+    b862-cap5                     8
+    b851-cap9                     8
+    b802-cap8                     8
+    b822-cap14                    8
+    b872-cap8                     8
+    b872-cap6                     8
+    b862-cap18                    8
+    b822-cap16                    8
+:x
```
Comment (tahigash) [inline src/genie/libs/parser/iosxe/tests/ShowApLedBrightnessLevelSummary/cli/equal/golden_output_output.txt:26]: garbage?

**Review by tahigash — DISMISSED** [review]

File: src/genie/libs/parser/iosxe/tests/ShowApDot115GhzChannel/cli/equal/golden_output_expected.py [hunk 8]
```diff
@@ -0,0 +1,28 @@
+expected_output = {
+    "channel_assignment": {
+        "chan_assn_mode": "AUTO",
+        "chan_upd_int": 12,
```
[the remark this answers is not in the document]
Comment (tahigash) [reply src/genie/libs/parser/iosxe/tests/ShowApDot115GhzChannel/cli/equal/golden_output_expected.py:4]: Please ignore. same added to https://github.com/CiscoTestAutomation/genieparser/pull/251

File: src/genie/libs/parser/iosxe/tests/ShowApDot115GhzChannel/cli/equal/golden_output_expected.py [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+        "channel_load": "Disable",
+        "device_aware": "Disable",
+        "clean_air": "Disabled",
+        "wlc_leader_name": "sj-00a-ewlc1",
+        "wlc_leader_ip": "10.7.5.133",
+        "last_run_seconds": 15995,
+        "dca_level": "MEDIUM",
+        "dca_db": 15,
+        "chan_width": "80 MHz",
+        "max_chan_width": 80,
```
[the remark this answers is not in the document]
Comment (tahigash) [reply src/genie/libs/parser/iosxe/tests/ShowApDot115GhzChannel/cli/equal/golden_output_expected.py:17]: please ignore. same added to https://github.com/CiscoTestAutomation/genieparser/pull/251

File: src/genie/libs/parser/iosxe/tests/ShowApDot115GhzChannel/cli/equal/golden_output_expected.py (hunk 5 above)
Comment (tahigash) [reply src/genie/libs/parser/iosxe/tests/ShowApDot115GhzChannel/cli/equal/golden_output_expected.py:16]: please ignore. same added to https://github.com/CiscoTestAutomation/genieparser/pull/251

File: src/genie/libs/parser/iosxe/tests/ShowApDot115GhzSummary/cli/equal/golden_output_expected.py (hunk 6 above)
Comment (tahigash) [reply src/genie/libs/parser/iosxe/tests/ShowApDot115GhzSummary/cli/equal/golden_output_expected.py:1]: please ignore. same added to https://github.com/CiscoTestAutomation/genieparser/pull/252

File: src/genie/libs/parser/iosxe/tests/ShowApLedBrightnessLevelSummary/cli/equal/golden_output_expected.py [hunk 10]
```diff
@@ -0,0 +1,64 @@
+expected_output = {
```
Comment (tahigash) [inline src/genie/libs/parser/iosxe/tests/ShowApLedBrightnessLevelSummary/cli/equal/golden_output_expected.py:1]: schema looks good

File: src/genie/libs/parser/iosxe/show_ap.py [hunk 11]
```diff
@@ -295,3 +295,680 @@ def cli(self, output=None):
...
+
+
+# ===============================
+# Schema for:
+#  * 'show ap dot11 5ghz channel'
+# ===============================
+class ShowApDot115GhzChannelSchema(MetaParser):
+    """Schema for show ap dot11 5ghz channel."""
+
+    schema = {
```
[the remark this answers is not in the document]
Comment (tahigash) [reply src/genie/libs/parser/iosxe/show_ap.py:307]: I believe this not scope of this PR.

File: src/genie/libs/parser/iosxe/show_ap.py (hunk 3 above)
Comment (tahigash) [reply src/genie/libs/parser/iosxe/show_ap.py:789]: this is not scope of PR.

File: src/genie/libs/parser/iosxe/show_ap.py (hunk 4 above)
Comment (tahigash) [reply src/genie/libs/parser/iosxe/show_ap.py:895]: reviewed. looks good.

File: src/genie/libs/parser/iosxe/show_ap.py (hunk 2 above)
Comment (qduk) [reply src/genie/libs/parser/iosxe/show_ap.py]: Updated

File: src/genie/libs/parser/iosxe/tests/ShowApLedBrightnessLevelSummary/cli/equal/golden_output_output.txt (hunk 7 above)
Comment (qduk) [reply src/genie/libs/parser/iosxe/tests/ShowApLedBrightnessLevelSummary/cli/equal/golden_output_output.txt]: Yes, Extraneous line removed

**Review by xiaoxinz-cisco — APPROVED** [review]

**Comment by xiaoxinz-cisco** [comment]
Hi @itdependsnetworks ,

Thank you for your contribution. It looks ready to merge once conflicts are resolved.

**Review by ThomasJRyan — APPROVED** [review]

**Pull request merged**